### ISSUE TITLE
Add option to add weight decay to neural network training

### ISSUE THREAD

**Issue by AmirAlavi** [AmirAlavi/scrna_nn#13]
Add l1 and l2 regularization options
